Docs 7997 ServiceNow 6.6.0 reference (#748)
mulesoft/docs-connectors: modules/ROOT/pages/servicenow/servicenow-reference-6.6.0.adoc

= ServiceNow Connector Reference - Mule 4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]

ServiceNow Connector Version 6.6

Anypoint Connector for ServiceNow (ServiceNow Connector) provides connections between Mule runtime engine (Mule) and ServiceNow apps.

Release Notes: xref:release-notes::connector/servicenow-connector-release-notes-mule-4.adoc[ServiceNow Connector Release Notes]

== Configurations

[[config]]
=== Config

> 

Reviewer: Is this because of the template, that the Configurations subtopic leads straight to Level-4 text?

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required

> |Name | String | The name for this configuration. Connectors reference the configuration with this name. | | x

Reviewer: ```suggestion
|Name | String | The name by which connectors reference the configuration. | | x
```

| Connection a| * <<config_basic-connection, Basic Authentication>> 

> * <<config_oauth2-auth-code-connection, OAuth 2.0 Authorization Code>> 

Reviewer: ditto caps?

 | The connection types to provide to this configuration. | | x

> | Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. This does not mean that the platform expires the instance at the exact moment that it becomes eligible. The runtime purges the instances when as appropriate. |  | 

Reviewer: Can we reduce the description to one sentence, something like...:
```suggestion
| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration (but not necessarily immediately purged).  |  | 
```

> | Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. This does not mean that the platform expires the instance at the exact moment that it becomes eligible. The runtime purges the instances when as appropriate. |  | 

Reviewer 2: "as appropriate" instead of "when as appropriate"

|===

==== Connection Types
[[config_basic-connection]]
===== Basic Authentication

> 

Reviewer: Again, not sure about these empty sections. I'm sure you know what you are doing; just mentioning it.

====== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required

> | Service Address a| String |  Address of ServiceNow Service. |  | x

Reviewer: Is | | empty for default value? If so, suggest "Not applicable."

Reviewer: DAGNABBIT. Forgot I wasn't supposed to edit. Nevermind!

| ServiceNow Version a| Enumeration, one of:

** EUREKA
** FUJI
** GENEVA
** HELSINKI
** ISTANBUL
** JAKARTA
** KINGSTON
** MADRID
** NEW_YORK
** ORLANDO |  ServiceNow API Version. |  ORLANDO | 
| User Tables List a| Array of String |  List of user-defined tables |  | 
| User Name a| String |  User Name |  | x
| Password a| String |  Password |  | x
| Transport Configuration a| One of:

* <<http-requester-based-transport-provider>>
* <<basic-auth-http-message-dispatcher-provider>> |  |  | 
| Sign Security Strategy a| <<WssSignSecurityStrategyAdapter>> |  A sign WSS configuration |  | 
| Username Token Security Strategy a| <<WssUsernameTokenSecurityStrategyAdapter>> |  A usernameToken WSS configuration |  | 

> | Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy |  | 

Reviewer 2: Need a period after the second sentence. (Global for this parameter)

| Custom Transport Configuration a| <<oauth2-http-message-dispatcher-provider>> |  |  | 
| Sign Security Strategy a| <<WssSignSecurityStrategyAdapter>> |  A sign WSS configuration |  | 
| Username Token Security Strategy a| <<WssUsernameTokenSecurityStrategyAdapter>> |  A usernameToken WSS configuration |  | 
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy |  | 
| Consumer Key a| String |  The OAuth consumerKey as registered with the service provider |  | x
| Consumer Secret a| String |  The OAuth consumerSecret as registered with the service provider |  | x
| Authorization Url a| String |  The service provider's authorization endpoint URL |  https://{instance-id}.service-now.com/oauth_auth.do | 
| Access Token Url a| String |  The service provider's accessToken endpoint URL |  https://{instance-id}.service-now.com/oauth_token.do | 
| Scopes a| String |  The OAuth scopes to be requested during the dance. If not provided, it defaults to those in the annotation |  | 

> | Resource Owner Id a| String |  The resourceOwnerId which each component should use if it doesn't reference otherwise. |  | 

Reviewer 2: This is unclear but I don't know if you can fix it easily.

| Callback Path a| String |  The path of the access token callback endpoint |  | x
| Authorize Path a| String |  The path of the local http endpoint which triggers the OAuth dance |  | x
| External Callback Url a| String |  If the callback endpoint is behind a proxy or should be accessed through a non direct URL, use this parameter to tell the OAuth provider the URL it should use to access the callback |  | 
| Object Store a| String |  A reference to the object store that should be used to store each resource owner id's data. If not specified, runtime will automatically provision the default one. |  | 
|===

== Supported Operations
* <<invoke>> 
* <<unauthorize>> 

== Operations

[[invoke]]
=== Invoke
`<servicenow:invoke>`

This operation invokes  ServiceNOW SOAP operations. 

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Service a| String |  |  | x
| Operation a| String |  The operation name of the service |  | x

> | Show Reference Values a| String |  Choose which values are returned for reference fields.Possible options are to return either sys_ids, display values or both.This option should only be used for get and getRecords operations. |  | 

Reviewer 2: Missing a space after the first sentence.

| Max Idle a| Number | Controls the maximum number of Mule components that can sit idle in the pool at any time. When set to a negative value, there is no limit to the number of Mule components that may be idle at one time. |  | 
| Max Wait a| Number | Specifies the number of milliseconds to wait for a pooled component to become available when the pool is exhausted and the exhaustedAction is set to WHEN_EXHAUSTED_WAIT. |  | 
| Min Eviction Millis a| Number | Determines the minimum amount of time an object may sit idle in the pool before it is eligible for eviction. When non-positive, no objects will be evicted from the pool due to idle time alone. |  | 
| Eviction Check Interval Millis a| Number | Specifies the number of milliseconds between runs of the object evictor. When non-positive, no object evictor is executed. |  | 
| Exhausted Action a| Enumeration, one of:

** WHEN_EXHAUSTED_GROW
** WHEN_EXHAUSTED_WAIT
** WHEN_EXHAUSTED_FAIL a| Specifies the behavior of the Mule component pool when the pool is exhausted:

* `WHEN_EXHAUSTED_GROW` +

> Creates a new Mule instance and returns it, which makes `maxActive` meaningless.

Reviewer 2: "ineffective" instead of "meaningless"?
